### ISSUE TITLE
Rename Label component to something more descriptive

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#1660]
[`<Label>`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/components/label/index.js) is a small component that allows passing a text and a screen reader text as props and takes care of displaying them with the correct classes and aria values.

Initially it was used instead of the HTML `<label>` tag, but we ended up using it in several other use cases. It would be good to rename it and its props to something more descriptive.

**Comment by mikejolley** [comment]
This component has since moved to the blocks-checkout package where it has been typed. So this should be solved.
